lightning/src/util/mut_global.rs

```diff
@@ -0,0 +1,73 @@
+//! An unsafe global variable that can be set exactly once.
+//!
+//! Used for testing purposes only.
+
+use std::sync::Mutex;
+
+/// A global variable that can be set exactly once.
+pub struct MutGlobal<T> {
+	value: Mutex<Option<T>>,
+}
+
+impl<T: Clone> MutGlobal<T> {
+	/// Create a new `MutGlobal` with no value set.
+	pub const fn new() -> Self {
+		Self {
+			value: Mutex::new(None),
+		}
+	}
+
+	/// Set the value of the global variable.
+	///
+	/// Ignores any attempt to set the value more than once.
+	pub fn set(&self, value: T) {
+		let mut lock = self.value.lock().unwrap();
+		if lock.is_none() {
```

Comment:
maybe we **should** allow multiple `set` calls - the caller can be responsible for any issues with different threads wanting different values.